packages/terra-abstract-modal/src/AbstractModal.jsx

```diff
@@ -100,22 +102,28 @@ const AbstractModal = (props) => {
   }, []);
 
   useEffect(() => {
-    function handleKeydown(e) {
+    function handleDocumentKeydown(e) {
       if (e.keyCode === KEY_ESCAPE && closeOnEsc && isOpen) {
-        if (modalElementRef.current) {
-          const body = document.querySelector('body');
-          if (e.target === modalElementRef.current || modalElementRef.current.contains(e.target) || e.target === body) {
-            onRequestClose();
-          }
+        const body = document.querySelector('body');
+        if (e.target === body) {
+          onRequestClose();
         }
       }
     }
-
-    document.addEventListener('keydown', handleKeydown);
-
+    document.addEventListener('keydown', handleDocumentKeydown);
     return (() => {
-      document.removeEventListener('keydown', handleKeydown);
+      document.removeEventListener('keydown', handleDocumentKeydown);
     });
+  }, [closeOnEsc, isOpen, onRequestClose]);
+
+  const handleKeydown = useCallback(e => {
+    if (e.keyCode === KEY_ESCAPE && closeOnEsc && isOpen) {
+      if (modalElementRef.current) {
```

Comment:
not a huge deal, but we could condense this if statement into the lower one:
```suggestion
      const currentModalNode = modalElementRef.current;
      if (currentModalNode && (e.target === currentModalNode|| currentModalNode.contains(e.target))) {
```